Leydis Laura Hernández Martínez C-211

leydis26 opened the pull request:
Estrategias implementadas: SJF; STCF; Round Robin; MLFQ

leydis26 committed:
Estrategia SJF implementada

Nueva función sjf_scheduler implementada. Modificaciones realizadas a la función fifo_scheduler, para verificar lel valor de cada parámetro en tiempo de ejecución. Casos de prueba: 001, 077

leydis26 committed:
Merge branch 'Sistemas-Operativos-Matcom:main' into main

leydis26 committed:
Merge branch 'Sistemas-Operativos-Matcom:main' into main

leydis26 committed:
Estrategia Round Robin implementada

Header incluido para utilizar el tipo de dato bool.
Las implementación para Round Robin y STCF no tienen en cuenta si el proceso seleccionado en cada momento aprovecha el CPU o se encuentra en una operación de I/O

leydis26 committed:
Estrategia MLFQ implementada

Modificado: round_robin_scheduler; get_scheduler

leydis26 committed:
Merge branch 'Sistemas-Operativos-Matcom:main' into develop

jmorgadov commented:
✅ - SJF
✅ - STCF
✅ - RR
⚠️ - MLFQ

MLFQ se me queda al parecer en un loop infinito y no está funcionando.

Nota propuesta: **3**

Puedes arreglar MLFQ y subir a 4 o 5 en dependencia de la correctitud de la implementación.

leydis26 committed:
Método auxiliar de la estrategia MLFQ modificado

modificado: priority_update
solucionado problema del loop infinito
Nuevo: método auxiliar para verificar en tiempo real el comportamiento de la estrategia MLFQ

leydis26 commented:
El problema del loop infinito ya está solucionado.
En el caso de prueba 59 el scheduler devuelve: Invalid PID = 12
Estado actual: Debugging
Faltan cambios para solicitar una nueva revisión

leydis26 committed:
MLFQ arreglado

leydis26 commented:
Proyecto listo para nueva revisión

Se agregaron métodos auxiliares para:
- saber en cada timer-interrupt;los procesos que no utilizanl CPU
- Dado una cola de prioridad verificar que todos sus elementos sean PIDs de procesos activos en el array procs_info

Se modificaron métodos auxiliares para tener en cuenta el comportamiento de los procesos que no utilizan el CPU en un timer-interrupt y garantizar que el PID devuelto por el scheduler en la estrategia MLFQ utiliza el CPU.

jmorgadov commented:
✅ - SJF
✅ - STCF
✅ - RR
✅ - MLFQ

Nota: **5**
